temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowTaskHandler.java

```diff
@@ -247,13 +248,17 @@ private Result createCompletedWFTRequest(
       }
       completedRequest.setStickyAttributes(attributes);
     }
+    WorkflowTaskCompletedMetadata.Builder metadata = WorkflowTaskCompletedMetadata.newBuilder();
     if (!result.getSdkFlags().isEmpty()) {
-      completedRequest =
-          completedRequest.setSdkMetadata(
-              WorkflowTaskCompletedMetadata.newBuilder()
-                  .addAllLangUsedFlags(result.getSdkFlags())
-                  .build());
+      metadata.addAllLangUsedFlags(result.getSdkFlags());
     }
+    if (!Strings.isNullOrEmpty(result.getWriteSdkName())) {
+      metadata.setSdkName(result.getWriteSdkName());
+    }
+    if (!Strings.isNullOrEmpty(result.getWriteSdkVersion())) {
+      metadata.setSdkVersion(result.getWriteSdkVersion());
+    }
+    completedRequest.setSdkMetadata(metadata.build());
```

Comment:
Will this set an empty workflow task completed metadata even if there are no SDK flags and no SDK name/version to set? If so I think this means it appears in every history JSON dump and technically takes a byte. I wonder if this whole chunk of code should be surrounded by `if (!result.getSdkFlags().isEmpty() || !Strings.isNullOrEmpty(result.getWriteSdkName()) || !Strings.isNullOrEmpty(result.getWriteSdkVersion())) {`.

Reply:
Yeah, it will do that, I'll change.